### PR TITLE
DOC Improve README and documentation

### PR DESCRIPTION
**Pull request by rth**
Re-organizes README and the documentation to hopefully be easier to navigate.
 
**README**

 - Move "building from source", and "testing" sections to the documentation as that's only relevant to a fraction of users and the long list of build dependencies at the top of the readme is scary.
 - Adds a link to "using pyodide from JS" under the "Getting started" section. Closes https://github.com/iodide-project/pyodide/issues/338
   @jeff-hykin do you find this updated readme easier to navigate? A rendered version can be found in https://github.com/rth/pyodide/tree/mv-changelog-contributing-docs
 - Remove navigation links on the top and bottom of the readme, and they are no longer necessary, I find that too many links is confusing, and starting with a bunch of links does not help looking at this readme on mobile.

**Documentation**

Also moved CHANGELOG.md and CONTRIBUTING.md to the documentation so it's in one place, and to reduce the number of files in the root folder.
 
We need to activate read-the-docs CI to be able to preview the documentation in PRs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,8 +6,6 @@
 
 The Python scientific stack, compiled to WebAssembly.
 
-[**Try Pyodide and Iodide in your browser**](https://alpha.iodide.io/notebooks/300/)
-
```
Comment (rth) [inline README.md:16]: Removed as there is already a section on "Try Pyodide without installing" below and a link to iodide in the following section.

File: docs/about.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# About the Project
+
+The Python scientific stack, compiled to WebAssembly.
+
+Pyodide brings the Python runtime to the browser via WebAssembly, along with
+the Python scientific stack including NumPy, Pandas, Matplotlib, parts of
+SciPy, and NetworkX. The [packages
+directory](https://github.com/iodide-project/pyodide/tree/master/packages)
+lists over 35 packages which are currently available.
```
Comment (rth) [inline docs/about.md:9]: This copies the about section while previously it was linked from the readme. The issue with the previous approach is that relative links can't work both in the documentation and the readme and TOC rendering was not great https://pyodide.readthedocs.io/en/latest/rootdir.html

This approach means these 3 paragraphs need to be maintained in sync between the doc and the readme but we have more flexibility.

File: docs/changelog.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Release notes
+
+## Unreleased
+
```
Comment (rth) [inline docs/changelog.md:4]: Will update the changelog in a separate PR.

**Comment by rth** [comment]
Yes, it's automatically re-generated. It's just that newer integration versions of readthedocs also add a CI status to PRs to inform that the build was successful, I think, but without it it's also fine.

**Comment by jeff-hykin** [comment]
> @jeff-hykin do you find this updated readme easier to navigate?

Thanks for working to address this so quickly, I appreciate it! 

The getting started is still a bit confusing to me. I might be misunderstanding the way the project works, but aren't the three bullet points only describing how to obtain pyodide.js and then the "using pyodide from Javascript" is describing how to usage? (regardless of whether you use a CDN, release download, or build from source)

The documentation inside "using pyodide from Javascript"  also talks about using a release (the 2nd bullet point) so it seems like it doesn't quite fit in the 1st bullet point. 


I think it would be sufficient to say:

> ## Getting Started
> First obtain `pyodide.js` from one of the sources
> 1. The CDN (link)
> 2. The release (link)
> 3. Building it from source (link)
>
> Once you have that you can use it in three different places
> 1. Here's the documentation on [using Pyodide from Javascript](https://pyodide.readthedocs.io/en/latest/using_pyodide_from_javascript.html) 
> 2. Here's the documentation on [using Pyodide from Iodide](https://pyodide.readthedocs.io/en/latest/using_pyodide_from_iodide.html)
> 3. Here's the documentation on [using Pyodide from a web worker](https://pyodide.readthedocs.io/en/latest/using_pyodide_from_webworker.html)

**Comment by rth** [comment]
It's a bit more complex than that. You don't need to download or build the sources to use Iodide: clicking on the link in the "try pyodide" section is enough.

Similarly using pyodide from webworker IMO doesn't need to be in the getting started guide. It's just a more complex use case of using it from JS. 

But overall yes, we should make things a bit more orthogonal between getting/ building, and using pyodide, both in the readme and the documentation.

**Pull request merged**